modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/PersistentPageMemoryStorageEngine.java

```diff
@@ -106,27 +110,29 @@ public void start() throws StorageException {
                     ? new AsyncFileIoFactory()
                     : new RandomAccessFileIoFactory();
 
-            partitionFilePageStoreManager = new PartitionFilePageStoreManager(
-                    Loggers.forClass(PartitionFilePageStoreManager.class),
+            filePageStoreManager = new FilePageStoreManager(
+                    Loggers.forClass(FilePageStoreManager.class),
                     igniteInstanceName,
                     storagePath,
                     fileIoFactory,
-                    ioRegistry,
                     pageSize
             );
 
-            partitionFilePageStoreManager.start();
+            filePageStoreManager.start();
         } catch (IgniteInternalCheckedException e) {
             throw new StorageException("Error starting file page store manager", e);
         }
 
+        partitionMetaManager = new PartitionMetaManager(ioRegistry, pageSize);
```

Comment:
Same here. Constructor initializes object, "start" starts all sub-components. This is how it was intended to work

Reply:
Understood nothing.

Reply:
Why do you have to assign values to fields in "start" and not in constructor?

Reply:
If page size it not known during the start, that would be a good answer for me. But even then we could just pass configuration instead of size.